helper/geojsonify.js

```diff
@@ -22,54 +24,30 @@ var DETAILS_PROPS = [
 ];
 
 
-var SOURCES = {
-  'geoname': 'gn',
-  'osmnode': 'osm',
-  'osmway': 'osm',
-  'admin0': 'qs',
-  'admin1': 'qs',
-  'admin2': 'qs',
-  'neighborhood': 'qs',
-  'locality': 'qs',
-  'local_admin': 'qs',
-  'osmaddress': 'osm',
-  'openaddresses': 'oa'
-};
-
 function lookupSource(src) {
-  return SOURCES.hasOwnProperty(src._type) ? SOURCES[src._type] : src._type;
+  var sources = type_mapping.type_to_source;
+  return sources.hasOwnProperty(src._type) ? sources[src._type] : src._type;
 }
 
+/*
+ * Use the type to layer mapping, except for Geonames, where having a full
+ * Elasticsearch document source allows a more specific layer name to be chosen
+ */
 function lookupLayer(src) {
-  switch(src._type) {
-    case 'osmnode':
-    case 'osmway':
-      return 'venue';
-    case 'admin0':
-      return 'country';
-    case 'admin1':
-      return 'region';
-    case 'admin2':
-      return 'county';
-    case 'neighborhood':
-      return 'neighbourhood';
-    case 'locality':
-      return 'locality';
-    case 'local_admin':
-      return 'localadmin';
-    case 'osmaddress':
-    case 'openaddresses':
-      return 'address';
-    case 'geoname':
-      if (src.category && src.category.indexOf('admin') !== -1) {
-        if (src.category.indexOf('admin:city') !== -1) { return 'locality'; }
-        if (src.category.indexOf('admin:admin1') !== -1) { return 'region'; }
-        if (src.category.indexOf('admin:admin2') !== -1) { return 'county'; }
-        return 'neighbourhood'; // this could also be 'local_admin'
-      }
-
-      if (src.name) { return 'venue'; }
-      if (src.address) { return 'address'; }
+  if (src._type === 'geoname') {
+    if (src.category && src.category.indexOf('admin') !== -1) {
+      if (src.category.indexOf('admin:city') !== -1) { return 'locality'; }
```

Comment:
Doh. I actually fully intended to do that (I even brought in lodash and used _.contains a few lines lower.